spec/request/route_destinations_spec.rb

```diff
@@ -590,6 +596,72 @@
       end
     end
 
+    describe 'Protocols' do
+      let(:params) do
+        {
+          destinations: [
+            {
+              app: {
+                guid: app_model.guid,
+                process: {
+                  type: 'assistant'
+                }
+              },
+              protocol: new_protocol
+            }
+          ]
+        }
+      end
+      let!(:existing_destination) do
+        VCAP::CloudController::RouteMappingModel.make(
+          app: app_model,
+          route: route,
+          process_type: 'assistant',
+          protocol: existing_protocol
+        )
+      end
+      context 'http1/http2' do
+        [
+          ['http2', 'http1', 'http1'],
+          ['http2', 'http2', 'http2'],
+          ['http2', nil,     'http1'],
```

Comment:
can you help me understand this case?  Why would `nil` change from `http2` to `http1`?  Would this happen if `protocol` was completely elided or only if it is explicitly set to `null`?